modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -1343,6 +1339,9 @@ def _get_dict_of_block_index(self, axis, indices, are_indices_sorted=False):
         if isinstance(indices, list):
             # Converting python list to numpy for faster processing
             indices = np.array(indices, dtype=np.int64)
+        # Fasttrack empty numpy array
+        if isinstance(indices, np.ndarray) and indices.size == 0:
+            return OrderedDict([(0, np.array([], dtype=np.int64))])
```

Comment:
I think we should add a comment explaining why this is needed.

Reply:
added